app/controllers/entries_controller.rb

```diff
@@ -32,6 +32,70 @@ def unread
     unread_entries = @user.unread_entries.select(:entry_id).page(params[:page]).sort_preference(@user.entry_sort)
     @entries = Entry.entries_with_feed(unread_entries, @user.entry_sort).entries_list
 
+    # Proper location for adjustments to unread entries?
+    # 
+    # structured as case to support future additional alternative
+    # post-query shuffle / sort options
+
+    case @user.entry_sort
+     when 'SHUFF'
+
+      # Current Feed Shuffling Algorithm
+      # 
+      # * Split retrieved entries by feed: O(n) time, O(n) space
+      # * Find the amount of entries in each feed (feed_queue_lengths): O(n) time
+      # * Establish frontier of most recent entry from each feed (frontier)
+      # * Find the time since publication of each item on the frontier (frontier_times)
+      # 
+      # * Produce score for each entry on frontier that is calculated as:
+      # *   a*t + b*(e^g)
+      #    
+      # *   a = adjustment factor for time since publication
+      # *   t = time since publication (in hours)
+      # *   b = adjustment factor for previous entries displayed
+      # *   e = number of previous entries from selected feed (in entries ^ growth factor)
+      # 
+      # * Add the first entry from the feed with the lowest score
+      # * Repeat through all unread entries O(# of feeds * n) 
+      # => ~ O(n) if n >> # of feeds ; O(n^2) if n ~= # of feeds
+
+      split_entries = {}
+    
+      @entries.map do |e| 
+        split_entries[e.feed_id] = [] unless split_entries[e.feed_id]
+        split_entries[e.feed_id].to_a.append e
+      end
+
+      feed_queue_lengths = Hash[ split_entries.keys.map { |k| split_entries[k].length } ]
```

Comment:
Yes - I thought I would need it, but it ended up that my implementation only needed @entries.length.